### PR TITLE
Throw warning when using DDP without DistributedSampler

### LINKED ISSUE #81: Relax requirement for DistributedSampler with ddp
**Is your feature request related to a problem? Please describe.**
I have an application where I'm using a custom `BatchSampler` to construct batches for the N-Pairs metric learning loss. I need all of the data to be available on all processes when using `DistributedDataParallel`, so I wouldn't want to use `DistributedSampler`, even if it was compatible with a custom `BatchSampler`. Right now, I've hit a wall because lightning throws this exception:

```
pytorch_lightning.utilities.debugging.MisconfigurationException: 
when using multiple gpus and multiple nodes you must pass
 a DistributedSampler to DataLoader(sampler).

ie: this:
dataset = myDataset()
dataloader = Dataloader(dataset)

becomes:
dataset = myDataset()
dist_sampler = torch.utils.data.distributed.DistributedSampler(dataset)
dataloader = Dataloader(dataset, sampler=dist_sampler)
```

**Describe the solution you'd like**
Could this exception be turned into a warning? I'm all for letting the user know when they're violating best practices, but throwing an exception removes flexibility for advanced users.

**Describe alternatives you've considered**
I looked at using the dp backend, but that's not going to work because the n-pairs loss needs the entire batch to compute the loss. Splitting it into chunks breaks things.

If I'm understanding correctly, this is actually another limitation introduced by Lightning. In a usual `DataParallel` setting, the batch would be merged back together before computing the loss and everything would be fine.

### PR DESCRIPTION
**Pull request by neggert**
Previously, we'd throw an exception. As described in #81, there are occasional reasons for wanting to use DDP without a DistributedSampler, such as non-standard batch construction. This PR converts the exception to a warning.

Closes #81

### CODE REVIEW

**Comment by williamFalcon** [comment]
@neggert awesome!  you'll  need to  also update 1  of the tests which checks  for the  old exception.  look in tests/test_models.py
